### PR TITLE
Optimized no_missing_parents function

### PR DESCRIPTION
**Pull request by eleftherioszisis**
I changed the ```no_missing_parents``` function to a numpy only operation, because I was profiling the load_neuron due to extremely slow times of loading multiple cells and I realised that there is a bottleneck. 

``` python
%timeit load_neuron('C140201A1_-_Scale_x1.000_y1.025_z1.000_-_Clone_5.h5')
1 loops, best of 3: 6.55 s per loop      # before optimization
1 loops, best of 3: 1.07 s per loop      # after optimization
```

```
   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
        1    4.918    4.918    4.918    4.918 check.py:47(no_missing_parents)
        3    0.724    0.241    4.468    1.489 utils.py:59(make_tree)
    60826    0.664    0.000    1.060    0.000 tree.py:79(ipreorder)
    60820    0.575    0.000    1.011    0.000 tree.py:217(as_elements)
    30423    0.574    0.000    1.196    0.000 hdf5.py:147(find_group)
        3    0.460    0.153    2.538    0.846 morphtree.py:94(find_tree_type)
        1    0.455    0.455    0.561    0.561 readers.py:122(__init__)
        1    0.418    0.418    2.357    2.357 hdf5.py:143(unpack_data)
60850/60849    0.374    0.000    1.863    0.000 utils.py:43(memoizer)
    30423    0.327    0.000    0.442    0.000 hdf5.py:158(find_parent_id)
    91558    0.315    0.000    0.315    0.000 :0(append)
    91613    0.306    0.000    0.306    0.000 :0(pop)
    60820    0.255    0.000    0.255    0.000 :0(hasattr)
    30438    0.239    0.000    0.340    0.000 readers.py:156(get_row)
    30423    0.235    0.000    0.454    0.000 fromnumeric.py:952(searchsorted)
    30423    0.220    0.000    0.220    0.000 :0(searchsorted)
    30423    0.215    0.000    0.215    0.000 readers.py:133(get_children)
    60820    0.201    0.000    0.201    0.000 :0(extend)
    30407    0.198    0.000    0.300    0.000 tree.py:49(add_child)
    60842    0.181    0.000    0.181    0.000 :0(isinstance)
```
I managed a speed up of approximately 5 times by removing the big internal time of the aforementioned function.
```
            3    0.498    0.166    3.037    1.012 utils.py:59(make_tree)
    60826    0.431    0.000    0.689    0.000 tree.py:79(ipreorder)
    60820    0.376    0.000    0.661    0.000 tree.py:217(as_elements)
    30423    0.369    0.000    0.797    0.000 hdf5.py:147(find_group)
        3    0.302    0.101    1.655    0.552 morphtree.py:94(find_tree_type)
60850/60849    0.294    0.000    1.288    0.000 utils.py:43(memoizer)
        1    0.272    0.272    1.613    1.613 hdf5.py:143(unpack_data)
    30438    0.226    0.000    0.295    0.000 readers.py:156(get_row)
    30423    0.214    0.000    0.288    0.000 hdf5.py:158(find_parent_id)
    91558    0.205    0.000    0.205    0.000 :0(append)
    91613    0.200    0.000    0.200    0.000 :0(pop)
    60820    0.166    0.000    0.166    0.000 :0(hasattr)
        1    0.162    0.162    0.228    0.228 readers.py:122(__init__)
    30423    0.149    0.000    0.291    0.000 fromnumeric.py:952(searchsorted)
    30423    0.147    0.000    0.147    0.000 readers.py:133(get_children)
    30423    0.142    0.000    0.142    0.000 :0(searchsorted)
    30423    0.138    0.000    0.138    0.000 :0(max)
    30407    0.135    0.000    0.202    0.000 tree.py:49(add_child)
    60820    0.132    0.000    0.132    0.000 :0(extend)
    60842    0.118    0.000    0.118    0.000 :0(isinstance)
```

### CODE REVIEW

File: neurom/io/check.py [hunk 1]
```diff
@@ -46,18 +46,16 @@ def has_sequential_ids(raw_data):
...
         tuple (bool, list of IDs that have no parent)
     '''
-    ids = raw_data.get_col(COLS.ID)
-    bad_ids = [int(ids[i]) for i in xrange(1, len(ids))
-               if raw_data.data_block[i][COLS.P] not in ids[:i]]
-
-    return len(bad_ids) == 0, bad_ids
+    ids = np.setdiff1d(raw_data.get_col(COLS.P),
+                       raw_data.get_col(COLS.ID))[1:]
+    return len(ids) == 0, list(ids.astype(np.int) + 1)
```
Comment (eleftherioszisis) [inline neurom/io/check.py]: Is it ok if a numpy array is returned instead of a list? I think there is no need to copy the data.

Comment (juanchopanza) [inline neurom/io/check.py]: I think that is fine. Do you have the results of `timeit` after the changes?

Comment (eleftherioszisis) [inline neurom/io/check.py]: Yes, in the initial post, the two timeit are before and after the changes with their respective profiles. ```no_missing_parent``` func is no more on the top of the profiling with respect to the internal time.

**Comment by juanchopanza** [comment]
Looks good to me. I can merge if you're ready.

**Comment by eleftherioszisis** [comment]
sure, feel free!

**Pull request merged**